packages/rulesets/src/spectral/az-common.ts

```diff
@@ -424,6 +427,38 @@ const ruleset : any = {
       "then": {
         "function": versionpolicy
       }
+    },
+    "az-default-in-enum": {
+      "description": "This rule applies when the value specified by the default property does not appear in the enum constraint for a schema.",
+      "message": "Default value should appear in the enum constraint for a schema",
+      "severity": "error",
+      "formats": [oas2],
+      "given": "$..[?(@.enum)]",
```

Comment:
this rule should apply to unresolved swagger too

Reply:
the matched enum can be a property name , like 
foo: {
   properties: {
     "enum": {
        type: string
        enum : ["1", "2"]
     }
 }
}

 should use more strict condition to match  to match the second enum

Reply:
I will check the type inside of function "defaultInEnum"